### ISSUE TITLE
Full static payload.js not using publicPath

### ISSUE THREAD

**Issue by scferg** [nuxt/nuxt.js#7923]
### Versions

- nuxt: 2.14.1
- node: 6.4.1

### Reproduction

nuxt.config.js below in Additional Details.

<details>
<summary>Additional Details</summary>
<br>

```
export default {
  /*
  ** Nuxt rendering mode
  ** See https://nuxtjs.org/api/configuration-mode
  */
  mode: 'universal',
  /*
  ** Nuxt target
  ** See https://nuxtjs.org/api/configuration-target
  */
  target: 'static',
  /*
  ** Headers of the page
  ** See https://nuxtjs.org/api/configuration-head
  */
  head: {
    title: process.env.npm_package_name || '',
    meta: [
      { charset: 'utf-8' },
      { name: 'viewport', content: 'width=device-width, initial-scale=1' },
      { hid: 'description', name: 'description', content: process.env.npm_package_description || '' }
    ],
    link: [
      { rel: 'icon', type: 'image/x-icon', href: '/favicon.ico' },
      { rel: 'stylesheet', href: 'https://use.typekit.net/tzk2btd.css' }
    ],
    script: [
      { src: 'https://unpkg.com/ionicons@5.1.2/dist/ionicons.js', body: true }
    ]
  },
  /*
  ** Global CSS
  */
  css: [
    '~/assets/styles/main.scss'
  ],
  styleResources: {
    scss: [
      '~/assets/styles/_variables.scss',
      'bootstrap/scss/_functions.scss',
      'bootstrap/scss/_variables.scss',
      'bootstrap/scss/_mixins.scss',
    ]
  },
  /*
  ** Plugins to load before mounting the App
  ** https://nuxtjs.org/guide/plugins
  */
  plugins: [
  ],
  /*
  ** Auto import components
  ** See https://nuxtjs.org/api/configuration-components
  */
  components: true,
  /*
  ** Nuxt.js dev-modules
  */
  buildModules: [
    '@nuxtjs/style-resources',
  ],
  /*
  ** Nuxt.js modules
  */
  modules: [
    // Doc: https://bootstrap-vue.js.org
    'bootstrap-vue/nuxt',
  ],
  /*
  ** Build configuration
  ** See https://nuxtjs.org/api/configuration-build/
  */
  build: {
    publicPath: 'https://khn.s3.us-east-1.amazonaws.com/virtual-series/'
  },
  generate: {
    subFolders: false
  },
  vue: {
    config: {
      ignoredElements: [/^ion-/]
    }
  },
  bootstrapVue: {
    bootstrapCSS: false,
    bootstrapVueCSS: false
  },
}

```
</details>

### Steps to reproduce

1. Specify a `publicPath` in nuxt.config.js for a full static app
2. Run `nuxt generate`
3. Deploy `/dist/` directory to static hosting (S3 in this case)
4. Visit static page and view console for incorrect URL errors for the `payload.js` file

### What is Expected?

I would expect the `payload.js` file to also use the specified `publicPath` instead of using a relative path. I added a `publicPath` specifically because I need absolute paths for everything.

### What is actually happening?

The `payload.js` files are being linked using relative paths. Bootstrap Vue actions (such as opening modals) are not functioning when deployed. I'm guessing it's related to this.

- `runtime.js` (correct): `https://khn.s3.us-east-1.amazonaws.com/virtual-series/runtime.ad4e919.js`
- `payload.js` (incorrect): `/static/1597456812/<page-filename>/payload.js`

All other assets are using the `publicPath` as expected. Am I missing something?

**Comment by asafyish** [comment]
I can confirm it's also not working for me. Every other resource does use the publicPath correctly.

**Comment by jackframework** [comment]
It's not working for me too.

**Comment by Grugier** [comment]
Same problem

**Comment by paullegrand** [comment]
+1, same problem

**Comment by r14c** [comment]
we are seeing this issue in `nuxt@2.14.6`

**Comment by pi0** [comment]
This should be fixed in [v2.14.8](https://github.com/nuxt/nuxt.js/releases/tag/v2.14.8). Please reopen if not.
